src/NServiceBus.Core.Analyzer/Handlers/AddHandlerInterceptor.Parser.cs

```diff
@@ -130,6 +185,36 @@ static bool IsHandlerInterface(INamedTypeSymbol type) => type is
             return new HandlerSpec(InterceptLocationSpec.From(location), handlerType.Name, handlerFullyQualifiedName, registrations);
         }
 
+        static IEnumerable<INamedTypeSymbol> GetTypeHierarchy(INamedTypeSymbol type, MarkerTypes markers)
+        {
+            var current = type.BaseType;
+            while (current is { SpecialType: not SpecialType.System_Object })
+            {
+                yield return current;
+                current = current.BaseType;
+            }
+
+            foreach (var iface in type.AllInterfaces)
+            {
+                if (!markers.IsMarkerInterface(iface))
+                {
+                    yield return iface;
+                }
+            }
+        }
+
+        class MarkerTypes(Compilation compilation)
```

Comment:
I have seen source generator where these types of marker types gets combined into the incremental pipeline from the beginning to cache them properly Is there any point in doing that too?

See for example

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs#L32-L33 and https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs#L43

Reply:
I think possibly but it screws up the ability to do the transform in the same way, so I'm thinking follow-up PR.